### PR TITLE
Add continuous integration

### PR DESCRIPTION
**Pull request by lilyminium**
I've added CI for pull requests to the `gh-pages` branch (not for pushes to the actual branch, since I think that is taken care of by the actual deployment -- let me know if I should change this). #2 is an example of failing CI from deleting all the posts.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+on:      
+  pull_request:
+    branches:
+      - gh-pages
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (davidlmobley) [inline .github/workflows/ci.yaml]: What does this do? The `actions/checkout@v2` portion, that is. Not used to the `@` syntax in these yet. (More familiar with Travis-CI tests).

**Review by davidlmobley — APPROVED** [review]
In general looks good; have you verified that failing builds fail the test? :) 

Can we add something at the top-level that notes that CI via GitHub actions is enabled? 

Otherwise looks good. Thanks so much!

**Comment by davidlmobley** [comment]
Ah, nevermind, you already noted that failing builds fail. Yay. This is gonna be great. :)

**Comment by davidlmobley** [comment]
I should probably make `gh-pages` protected, then... ?

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (lilyminium) [reply .github/workflows/ci.yaml]: It checks out version 2 of the actions/checkout action -- I commented something to that effect. You can also pin to a specific commit or tag :)

**Comment by lilyminium** [comment]
> Can we add something at the top-level that notes that CI via GitHub actions is enabled?

I added a badge button and also running CI on push to gh-pages, if that helps?

> I should probably make `gh-pages` protected, then... ?

If you think reviews are necessary, sure. It can also become an unnecessary block to publishing if no one has time to review, though.

**Comment by davidlmobley** [comment]
OK, thanks, merging!!

**Pull request merged**